be1-go/channel/lao/mod.go

```diff
@@ -377,78 +321,133 @@ func (c *Channel) processLaoState(rawMessage message.Message, msgData interface{
+	var rollCallOpen messagedata.RollCallOpen
 
-	err := msg.UnmarshalData(&witnessData)
+	err := msg.UnmarshalData(&rollCallOpen)
 	if err != nil {
-		return xerrors.Errorf("failed to unmarshal witness data: %v", err)
+		return xerrors.Errorf("failed to unmarshal roll call open: %v", err)
 	}
 
-	err = schnorr.VerifyWithChecks(crypto.Suite, []byte(msg.Sender), []byte(witnessData.MessageID), []byte(witnessData.Signature))
+	// check that data is correct
+	err = c.verifyMessageRollCallOpen(rollCallOpen)
 	if err != nil {
-		return answer.NewError(-4, "invalid witness signature")
+		return xerrors.Errorf("invalid roll_call#open message: %v", err)
 	}
 
-	err = c.inbox.AddWitnessSignature(witnessData.MessageID, msg.Sender, witnessData.Signature)
-	if err != nil {
-		return xerrors.Errorf("failed to add witness signature: %w", err)
+	if !c.rollCall.checkPrevID([]byte(rollCallOpen.Opens)) {
+		return answer.NewError(-1, "The field `opens` does not correspond to the id of "+
+			"the previous roll call message")
 	}
 
+	c.rollCall.id = rollCallOpen.UpdateID
+	c.rollCall.state = Open
+
 	return nil
 }
 
-func (c *Channel) createChirpingChannel(publicKey string, socket socket.Socket) {
-	chirpingChannelPath := c.channelID + social + publicKey
+// processRollCallClose processes a close roll call message.
+func (c *Channel) processRollCallClose(msg message.Message, msgData interface{},
+	senderSocket socket.Socket) error {
 
-	cha := chirp.NewChannel(chirpingChannelPath, publicKey, c.hub, c.general, be1_go.Logger)
-	c.hub.NotifyNewChannel(chirpingChannelPath, cha, socket)
-	log.Info().Msgf("storing new chirp channel (%s) for: '%s'", c.channelID, publicKey)
+	data, ok := msgData.(*messagedata.RollCallClose)
+	if !ok {
+		return xerrors.Errorf("message %v isn't a rollcall#close message", msgData)
+	}
+
+	// check that data is correct
+	err := c.verifyMessageRollCallClose(data)
+	if err != nil {
+		return xerrors.Errorf("invalid roll_call#close message: %v", err)
+	}
+
+	if c.rollCall.state != Open {
+		return answer.NewError(-1, "The roll call cannot be closed since it's not open")
+	}
+
+	if !c.rollCall.checkPrevID([]byte(data.Closes)) {
+		return answer.NewError(-4, "The field `closes` does not correspond to the id of "+
+			"the previous roll call message")
+	}
+
+	c.rollCall.id = data.UpdateID
+	c.rollCall.state = Closed
+
+	var db *sql.DB
+
+	if sqlite.GetDBPath() != "" {
+		db, err := sql.Open("sqlite3", sqlite.GetDBPath())
+		if err != nil {
+			c.log.Err(err).Msg("failed to connect to db")
+			db = nil
+		} else {
+			defer db.Close()
+		}
+	}
+
+	for _, attendee := range data.Attendees {
+		c.attendees[attendee] = struct{}{}
+
+		c.createChirpingChannel(attendee, senderSocket)
+
+		c.reactions.AddAttendee(attendee)
+
+		if db != nil {
+			c.log.Info().Msgf("inserting attendee %s into db", attendee)
+
+			err := insertAttendee(db, attendee, c.channelID)
+			if err != nil {
+				c.log.Err(err).Msgf("failed to insert attendee %s into db", attendee)
```

Comment:
In my opinion it is only a log and not really an error that will stop the correct behavior of the backend. Other similar logs are made the same way a bit earlier. 

I don't even know if the database feature works really. (Actually I didn't even know it existed)

Reply:
- the DB is not working and should eventually be removed from the codebase
- there is a log and not a return because we don't want the DB operation to block the rest of the execution in case of an error